common/canonical_serialization/src/canonical_serialize.rs

```diff
@@ -64,7 +64,16 @@ pub trait CanonicalSerializer {
     fn encode_btreemap<K: CanonicalSerialize, V: CanonicalSerialize>(
         &mut self,
         v: &BTreeMap<K, V>,
-    ) -> Result<&mut Self>;
+    ) -> Result<&mut Self> {
+        self.encode_tuple_iterator(v.iter())
+    }
+
+    fn encode_tuple_iterator<K: CanonicalSerialize, V: CanonicalSerialize, I>(
+        &mut self,
+        iter: I,
```

Comment:
Uhh, this stinks, I really miss OO code...

Reply:
If `iter: impl Iterator<Item = (K, V)>` works this might look a bit better.